build/helper/metadata_add_all.py

```diff
@@ -297,6 +297,15 @@ def _fix_type(parameter):
     parameter['type'] = parameter['type'].replace('[ ]', '[]').replace(' []', '[]').replace(' ', '_')
 
 
+def _fix_custom_type(parameter, config):
```

Comment:
Please add a comment here, along the lines of:

> TODO(olsl21): Metadata is inconsistent with regards to how structs are treated.
> This is a hack to workaround that inconsistency. The real root caused is tracked
> by internal NI bug 1918101. Once that is addressed, this method can be removed.

Reply:
I've added the TODO comment in the latest commit.